department-of-veterans-affairs/abd-vro
# Save requests to Redis

## What was the problem?


For error recovery scenarios, we cannot "replay" request submitted to VRO's API b/c we don't currently have approval to persist PII data.

Associated tickets or Slack threads:
- #563

## How does this fix it?[^1]


Log the payload of incoming requests to in-memory (non-persisted) Redis so that we can manually resubmit those request when needed.

## How to test this PR
- `export COMPOSE_PROFILES=debug` -- the console container only starts if the docker-compose `debug` profile is enabled.
- Start VRO: `./gradlew app:dockerComposeUp`. Note the `vro-console-1` container (optionally check out the logs: `docker logs vro-console-1`).
- Using Swagger, test the 2 VRO endpoints
- Attach to the container: `docker attach vro-console-1` (Alternative: use `redis-cli` to run equivalent commands.)
- On the `groovy:000>` prompt, try the following:
```groovy
// List all the Redis keys available
groovy:000> redis.keys "*"

// "1234" is the claim id
groovy:000> redis.hgetall "tracking-1234"
// Shows the keys and values of the hash at Redis key "tracking-1234"

// Query the individual hash entries
groovy:000> redis.hget "tracking-1234", "submitted-claim"
groovy:000> redis.hget "tracking-1234", "submitted-pdf"
```

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
